File: example/assets/lustre/src/ffi/lissome_ffi.erl
-module(lissome_ffi).
-export([
  erlang_version/0,
  atoms_count/0,
  ports_count/0,
  processes_count/0,
  uptime/0,
  total_memory_usage/0
]).

erlang_version() ->
  unicode:characters_to_binary(erlang:system_info(system_version)).

atoms_count() ->
  erlang:system_info(atom_count).

ports_count() ->
  erlang:system_info(port_count).

processes_count() ->
  erlang:system_info(process_count).

uptime() ->
  Uptime = erlang:statistics(wall_clock),
  erlang:element(1, Uptime).

total_memory_usage() ->
  {total, TotalMemory} = lists:keyfind(total, 1, erlang:memory()),
  TotalMemory.

File: example/assets/lustre/src/ffi/lissome_ffi.ex
defmodule LissomeFFI do
  def erlang_version, do: List.to_string(:erlang.system_info(:system_version))

  def atoms_count, do: :erlang.system_info(:atom_count)

  def ports_count, do: :erlang.system_info(:port_count)

  def processes_count, do: :erlang.system_info(:port_count)

  def uptime, do:
   round(elem(:erlang.statistics(:wall_clock), 0) / 60)

  def total_memory_usage, do: :erlang.memory()[:total]
end
